### PR TITLE
Update to SDL 2.0.20

### PR DESCRIPTION
**Pull request by InfoTeddy**
We have a couple things from SDL 2.0.18 that really make our life easier, currently hidden behind a couple ifdefs. In the meantime, 2.0.20 also quickly released as well, so we're doing that too.

## Legal Stuff:

By submitting this pull request, I confirm that...

- [X] My changes may be used in a future commercial release of VVVVVV
- [X] I will be credited in a `CONTRIBUTORS` file and the "GitHub Friends"
  section of the credits for all of said releases, but will NOT be compensated
  for these changes

### CODE REVIEW

**Comment by flibitijibibo** [comment]
Once brew and vcpkg are updated this is good to go.

**Review by flibitijibibo — APPROVED** [review]

**Comment by flibitijibibo** [comment]
brew is now updated. vcpkg seems to require someone manually filing a PR, unless there's some update notifications they have that I can't see from here.

**Comment by flibitijibibo** [comment]
This PR should be updated after https://github.com/microsoft/vcpkg/pull/21848 is merged.

**Comment by flibitijibibo** [comment]
Looks like the holdup on vcpkg's end was a UWP issue. They're supposed to be sending a patch for 2.0.20, which is currently set for early January due to an emergency regression in 2.0.18.

**Comment by flibitijibibo** [comment]
2.0.20 has been released.

brew has updated to 2.0.20, vcpkg is trying to update once again https://github.com/microsoft/vcpkg/pull/22586

**Comment by flibitijibibo** [comment]
Oh yeah, forgot about that - yes, let's move to the GitHub archives ASAP.

**Comment by flibitijibibo** [comment]
vcpkg has updated to 2.0.20. Let's update this patchset and move to the GitHub binaries later.

File: desktop_version/Dockerfile [hunk 1]
```diff
@@ -2,9 +2,9 @@ FROM centos:7
 
 # run first to improve caching (other things update more often than SDL2)
 WORKDIR /tmp
-RUN curl -O https://www.libsdl.org/release/SDL2-2.0.16.tar.gz
-RUN tar -xf SDL2-2.0.16.tar.gz
-RUN mkdir SDL2-2.0.16/build
+RUN curl -O https://www.libsdl.org/release/SDL2-2.0.18.tar.gz
+RUN tar -xf SDL2-2.0.18.tar.gz
+RUN mkdir SDL2-2.0.18/build
```
Comment (flibitijibibo) [inline desktop_version/Dockerfile:7]:
```suggestion
RUN curl -O https://www.libsdl.org/release/SDL2-2.0.20.tar.gz
RUN tar -xf SDL2-2.0.20.tar.gz
RUN mkdir SDL2-2.0.20/build
```

File: desktop_version/Dockerfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ RUN yum-builddep -y SDL2
 
 RUN yum clean all
 
-WORKDIR /tmp/SDL2-2.0.16/build
+WORKDIR /tmp/SDL2-2.0.18/build
```
Comment (flibitijibibo) [inline desktop_version/Dockerfile:28]:
```suggestion
WORKDIR /tmp/SDL2-2.0.20/build
```

File: desktop_version/Dockerfile [hunk 3]
```diff
@@ -36,6 +36,6 @@ RUN make -j $(nproc)
 RUN make install
 
 WORKDIR /tmp
-RUN rm -rf SDL2-2.0.16.tar.gz SDL2-2.0.16/ SDL2_mixer-2.0.4.tar.gz SDL2_mixer-2.0.4/
+RUN rm -rf SDL2-2.0.18.tar.gz SDL2-2.0.18/ SDL2_mixer-2.0.4.tar.gz SDL2_mixer-2.0.4/
```
Comment (flibitijibibo) [inline desktop_version/Dockerfile:39]:
```suggestion
RUN rm -rf SDL2-2.0.20.tar.gz SDL2-2.0.20/ SDL2_mixer-2.0.4.tar.gz SDL2_mixer-2.0.4/
```

File: desktop_version/README.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ To generate the projects on Windows:
 # Put your SDL2/SDL2_mixer folders somewhere nice!
 mkdir flibitBuild
 cd flibitBuild
-cmake -A Win32 -G "Visual Studio 10 2010" .. -DSDL2_INCLUDE_DIRS="C:\SDL2-2.0.16\include;C:\SDL2_mixer-2.0.4\include" -DSDL2_LIBRARIES="C:\SDL2-2.0.16\lib\x86\SDL2;C:\SDL2-2.0.16\lib\x86\SDL2main;C:\SDL2_mixer-2.0.4\lib\x86\SDL2_mixer"
+cmake -A Win32 -G "Visual Studio 10 2010" .. -DSDL2_INCLUDE_DIRS="C:\SDL2-2.0.18\include;C:\SDL2_mixer-2.0.4\include" -DSDL2_LIBRARIES="C:\SDL2-2.0.18\lib\x86\SDL2;C:\SDL2-2.0.18\lib\x86\SDL2main;C:\SDL2_mixer-2.0.4\lib\x86\SDL2_mixer"
```
Comment (flibitijibibo) [inline desktop_version/README.md:28]:
```suggestion
cmake -A Win32 -G "Visual Studio 10 2010" .. -DSDL2_INCLUDE_DIRS="C:\SDL2-2.0.20\include;C:\SDL2_mixer-2.0.4\include" -DSDL2_LIBRARIES="C:\SDL2-2.0.20\lib\x86\SDL2;C:\SDL2-2.0.20\lib\x86\SDL2main;C:\SDL2_mixer-2.0.4\lib\x86\SDL2_mixer"
```

**Review by flibitijibibo — COMMENTED** [review]
Applying this should trigger a rebuild, which _should_ pass...

**Comment by InfoTeddy** [comment]
I was already working on this again today. I still have to push the Docker container (already built it, just need to figure out how to push again...) unless you're fine with still using 2.0.18.

**Comment by flibitijibibo** [comment]
Build is happy again, and we're on 2.0.20. Will wait for the Docker stuff though.

**Pull request merged**